source/clear-linux/guides/maintenance/kernel-development.rst

```diff
@@ -0,0 +1,462 @@
+   configuration tool. Below are some popular options:
+
+   - :command:`$EDITOR .config` - the .config file can be directly edited for 
+     simple changes with names that are already known.
+
+   - :command:`make config` - a text-based tool that asks questions 
+     one-by-one to decide configuration options. 
+
+   - :command:`make menuconfig` - a terminal user interface that provides 
+     menus to decide configuration options.
+
+   - :command:`make xconfig` - a graphical user interface that provides 
+     tree views to decide configuration options.
+
+
+   More configuration tools can be found by looking at the make help: 
+   :command:`make help | grep config`
+
+
+#. Commit and save the changes to the :file:`.config` file.
+
+
+#. Copy the :file:`.config` file from the kernel source directory into 
+   the kernel package directory as :file:`config` for inclusion in the build.
+
+   .. code-block:: bash
+
+      cp .config ../config
+
+
+
+Modify kernel source code 
+=========================
+
+Changes to kernel code are applied with patch files. Patch files are 
+formatted git commits that can be applied to the main source code.
+
+You will need to obtain a copy of the source code,
+make modifications, generate patch file(s), and add them to the RPM SPEC 
+file for inclusion during the kernel build.
+
+If you have a large number of patches or a more complex workflow, 
+consider using a patch management tool in addition to Git such as 
+`Quilt`_. 
+
+
+#. Make sure you have followed the steps to 
+   `Pull a copy of the Linux kernel source code`_ and are in the kernel 
+   source working directory.
+
+
+#. Make any desired code changes to the Linux source code files.
+
+
+#. Track and commit your changes to the local git repo. 
+
+   .. code-block:: bash
+
+      git add <FILENAME>
+      git commit -m "My patch for driver A" <FILENAME>
+
+
+#. Generate a patch file based on your git commits. 
+   <n> represents the number of local commits to create patch file. 
+   See the `git-format-patch Documentation`_ for detailed information 
+   on using :command:`git format-patch`
+
+   .. code-block:: bash
+      
+      git format-patch -<n>
+
+
+#. Copy the patch files from the patches directory in the linux 
+   source tree to the RPM build directory.
+
+   .. code-block:: bash
+      
+      cp *.patch ~/clearlinux/packages/linux/
+
+
+#. Navigate back to the RPM build directory.
+
+   .. code-block:: bash
+      
+      cd ~/clearlinux/packages/linux/
+
+
+#. Open the Linux kernel package RPM SPEC file in an editor.
+
+   .. code-block:: bash
+
+      $EDITOR linux.spec
+
+
+#. Locate the section of the SPEC file that contains existing patch 
+   variable definitions and append your patch file name. Ensure the 
+   patch number does not collide with an existing patch. 
+   In this example, the patch file is called 
+   :file:`2001-my-patch-for-driver-A.patch`
+
+   .. code-block:: bash
```

Comment:
There is no mention on the document to apply the current CL patch set

Reply:
Can you clarify what you mean? In this example, readers would be using the existing RPM SPEC file which already has the CL patches. This is just a snippet.

Reply:
Yes, but the user did not apply any of those patches to the kernel source.

They should:

1. get the repo
2. get the kernel sources
3. apply CL patches to the kernel sources
4. copy `config` to kernel sources as `.config`
5. do development (any changes)
6. create the patch
7. add to the repo/spec file

Reply:
8. copy the modifed `.config` to repo `config`

Reply:
Thanks @miguelinux. As far as I can tell, only step 3 is missing to apply the CL patches to the copy of the kernel source directory. Is this required? If so, can you help explain why? Is it so that any user-generated patch files take into the account the CL patches?

Reply:
@miguelinux in any case, I have added step 3. Let me know if you think it looks good.

Reply:
if you want to modify the kernel from CL you need to have All the patches applied to the current kernel.
Some patches add Kconfig options, so when you use the `config` file as reference you could lose those features. 

If the kernel won't be based on CL so you don't need to apply the patches or use the config as base.